master/internal/api_trials.go

```diff
@@ -912,6 +912,68 @@ func (a *apiServer) streamMetrics(ctx context.Context,
 	return nil
 }
 
+// Create a TrialSourceInfo, which serves as a link between trials and checkpoints
+// used for tracking purposes for fine tuning and inference.
+func (a *apiServer) ReportTrialSourceInfo(
+	ctx context.Context, req *apiv1.ReportTrialSourceInfoRequest,
+) (*apiv1.ReportTrialSourceInfoResponse, error) {
+	// TODO (Taylor): Handle user auth/rbac
+	resp := &apiv1.ReportTrialSourceInfoResponse{}
+	tsi := req.TrialSourceInfo
+	query := db.Bun().NewInsert().Model(tsi).
+		Value("trial_source_info_type", "?", tsi.TrialSourceInfoType.String()).
+		Returning("trial_id").Returning("checkpoint_uuid")
+	if tsi.GetSourceModelVersionId() == 0 {
+		query.ExcludeColumn("source_model_version_id")
+	}
+	if tsi.GetSourceModelVersionVersion() == 0 {
+		query.ExcludeColumn("source_model_version_version")
+	}
```

Comment:
commented on proto: these fields should be made optional, and then 
```suggestion
	if tsi.SourceModelVersionId == nil {
		query.ExcludeColumn("source_model_version_id")
	}
	if tsi.SourceModelVersionVersion == nil {
		query.ExcludeColumn("source_model_version_version")
	}
```